README.md
![alt XOOPS CMS](http://xoops.org/images/logoXoops4GithubRepository.png)
## indexScan module for [XOOPS CMS 2.5.8+](https://xoops.org)
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](LICENSE) 
[![Scrutinizer Code Quality](https://img.shields.io/scrutinizer/g/XoopsModules25x/indexscan.svg?style=flat)](https://scrutinizer-ci.com/g/XoopsModules25x/indexscan/?branch=master)
[![Codacy Badge](https://api.codacy.com/project/badge/grade/166bbc6bf0934c9d8a34be46b36661dd)](https://www.codacy.com/app/mambax7/indexscan_2)
[![Code Climate](https://img.shields.io/codeclimate/github/XoopsModules25x/indexscan.svg?style=flat)](https://codeclimate.com/github/XoopsModules25x/indexscan)
[![SensioLabsInsight](https://insight.sensiolabs.com/projects/a1721ee0-7c5c-4854-8c83-38800c358607/mini.png)](https://insight.sensiolabs.com/projects/a1721ee0-7c5c-4854-8c83-38800c358607)
[![Latest Pre-Release](https://img.shields.io/github/tag/XoopsModules25x/indexscan.svg?style=flat)](https://github.com/XoopsModules25x/indexscan/tags/)
[![Latest Version](https://img.shields.io/github/release/XoopsModules25x/indexscan.svg?style=flat)](https://github.com/XoopsModules25x/indexscan/releases/)

**indexScan** module for [XOOPS CMS](http://xoops.org) is to scan your website folders for missing index.html files. If any of them is missing, you can create them automatically.

[![Tutorial Available](http://xoops.org/images/tutorial-available-blue.svg)](https://www.gitbook.com/book/xoops/indexscan-tutorial/) Tutorial: see [GitBook](https://www.gitbook.com/book/xoops/indexscan-tutorial/). 
To contribute to the Tutorial, [fork it on GitHub](https://github.com/XoopsDocs/indexscan-tutorial)

[![Translations on Transifex](http://xoops.org/images/translations-transifex-blue.svg)](https://www.transifex.com/xoops) 

Please visit us on  [http://xoops.org](http://xoops.org)

The upcoming "next generation" version of XOOPS CMS is being crafted on GitHub at: https://github.com/XOOPS



docs/changelog.txt
<h5>2.20 Beta 1   [WORK IN PROGRESS - NOT RELEASED]</h5>  Dev: XOOPS 2.5.11, PHP 7.4.15, PHP 8.0.2
<hr>
- updates, cosmetics (mamba)
- namespaces (mamba)
- changed from index.html to index.php (mamba)
- refactored functions from main.php into Utility (mamba)

2.10 Beta 1                2016-06-27
---------------------------------
- Converted to XOOPS 2.5.8 Admin GUI (mamba)
- update for PHP7 and XOOPS 2.5.8 (mamba)
- removed "@version" (mamba)
- standardization with addNavigation(basename(__FILE__)) (mamba)
- removed unneeded double quotes and parenthesis (mamba)
- improved type comparison (mamba)
- standardization with <br> (mamba)
- small changes based on testing tool feedback (mamba)
- code cosmetics (mamba)

2.1 Monday 18 January 2010 - 18:50 (gmt +1)
-------------------------------

- Added possibility to check files in web folders against xoops check file and filter from Config. Filter separates extensions of files presumed to be ok.
- Files found in check files marked red are presumed not to be part of XOOPS, and can be deleted by the 'delete icon' using ajax + jquery avoid reloading the page.
- Added the PclZip.Class to create zip file on the fly for users not having php > 5.20.
- Scan to create back up from the folders residing in the admin/folder2backup. The scan creates a copy of the empty folders, existing index.html, index.php, index.htm files, delete all other files, creates new indexfiles, and create link to download as zip. Handy when working with developing and want to create a zip with nothing but index files.
- I tested with htdoc from XOOPS 2.4.4 in a Wamp, creation time is aprox. 2 minutes. Not tested in real server yet, but time must be considerable.

2.02
-------------------------------------
- Added the option in config to change the base of the starting point for the scans. To avoid users - having xoops_data, xoops_lib outside of the root, not having all folders scanned.
- Added a function to get the $xoopsModuleConfig (root) where global $xoopsModuleConfig not working in admin.
- New language defines are.:

2.0 friday 27 november 2009 - 13:28 (gmt +1)
----------------------------------
- Added searchword "fromCharCode" to search word in iFrame scan, to look for encoded javascript includes
- Spiced the scanning up with some highlighting javascript + some jquery for the scan and messages with lazy load
- Excluded the modules/indexscan from the iframe scan. The description and functions contain multible iframe-words and thus always gets caught in the scan.

1.1 Wednesday 16 September 2009 - 13:30 (gmt +1)
-----------------------------
- Added the ability to scan for use of iFrame in indexfiles // Credit to Ghia for this Idea
- Added Spanish translation // Thanks to DonCurioso at hispaxoops.com


1.1 thursday 17 september 2009 - 21:00 (gmt +1)
-------------------------------
- Indexscan included words with (header, mainfile and footer) in the iFrame injection scan.


1.1 Friday 02 October 2009 - 21:30 (gmt +1)
-----------------------------------
- Changed xoops_version.php to show the correct release number 1.1 (thanks to Burning)
- Added a new and very cool logo for indexscan (million thanks again to Burning)



 1.0 Saturday 7 September 2009 - 15.55 (gmt +1)
----------------------------------------
- Added French Language to the module // Thanks to kris_fr

 1.0 Saturday 12 September 2009 - 10.00 (gmt +1)
 ------------------------------------
 - Added dutch translation thanks to Reintjan


 1.0 Thursday 9 july 2009 - 17:08 (GMT +1)
----------------------------------------
- Changed function to create index.html, as it ignored index.php3. Now working

 V1.0 Saturday 5 September 2009 - 14.00 (gmt +1)

- Added Persian Language to the module // Thanks to Voltan
- Decided to let the module go official. No errors or concerns has been reported during testing.


xoops_version.php
<?php
/**
 * Index Scan module
 *
 * Use this module to scan your web folders for missing index files. If any found
 * you can create automatically.
 *
 * The module uses a function to scan for files originally found on php.net examples
 * but modified to suit the needs / standards of xoops 2.3.3 & php5.
 *
 * LICENSE
 *
 * You may not change or alter any portion of this comment or credits
 * of supporting developers from this source code or any supporting source code
 * which is considered copyrighted (c) material of the original comment or credit authors.
 *
 * @copyright     The XOOPS Project http://sourceforge.net/projects/xoops/
 * @license       http://www.fsf.org/copyleft/gpl.html GNU public license
 * @author        Michael Albertsen (culex) <http://www.culex.dk>
 * @since         File available since Release 1.0.1
 */

$modversion['name']        = _MI_INDEXSCAN_MODULE_NAME;
$modversion['version']     = '2.10';
$modversion['description'] = _MI_INDEXSCAN_MODULE_DESC;
$modversion['credits']     = 'Developed by Culex http://www.culex.dk, Thanks XOOPS Community for helping out, translators for translating and especially Burning for The cool Logo';
$modversion['author']      = 'Culex';
$modversion['help']        = 'page=help';
$modversion['license']     = 'GNU GPL 2.0 or later';
$modversion['license_url'] = 'www.gnu.org/licenses/gpl-2.0.html';
$modversion['official']    = 1;
$modversion['image']       = 'logo_module.png';
$modversion['dirname']     = 'indexscan';

$modversion['dirmoduleadmin'] = 'Frameworks/moduleclasses/moduleadmin';
$modversion['sysicons16']     = 'Frameworks/moduleclasses/icons/16';
$modversion['sysicons32']     = 'Frameworks/moduleclasses/icons/32';
$modversion['modicons16']     = 'assets/images/icons/16';
$modversion['modicons32']     = 'assets/images/icons/32';
//about
$modversion['release_date']        = '2016/06/26';
$modversion['module_website_url']  = 'www.xoops.org';
$modversion['module_website_name'] = 'XOOPS';
$modversion['module_status']       = 'Beta 1';
$modversion['min_php']             = '5.5';
$modversion['min_xoops']           = '2.5.8';
$modversion['min_admin']           = '1.2';
$modversion['min_db']              = array(
    'mysql'  => '5.0.7',
    'mysqli' => '5.0.7'
);

// Sql file (must contain sql generated by phpMyAdmin or phpPgAdmin)
// All tables should not have any prefix!

// Tables created by sql file (without prefix!)

//Admin things
$modversion['hasAdmin']    = 1;
$modversion['system_menu'] = 1;
$modversion['adminindex']  = 'admin/index.php';
$modversion['adminmenu']   = 'admin/menu.php';

// Menu
$modversion['hasMain']        = 0;
$modversion['sub'][1]['name'] = _MI_INDEXSCAN_SCANNOW;
$modversion['sub'][1]['url']  = 'main.php?op=ScanNow';
$modversion['sub'][2]['name'] = _MI_INDEXSCAN_CREATEINDEX;
$modversion['sub'][2]['url']  = 'main.php?op=CreateNow';

// Blocks

// Templates

$modversion['config'][] = array(
    'name'        => 'indexscan_frombackup',
    'title'       => '_MI_INDEXSCAN_FROMBACKUP',
    'description' => '_MI_INDEXSCAN_FROMBACKUP_DESC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => 'testing'
);

$modversion['config'][] = array(
    'name'        => 'indexscan_rootorsub',
    'title'       => '_MI_INDEXSCAN_ROOTORSUB',
    'description' => '_MI_INDEXSCAN_ROOTORSUB_DESC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '../../../'
);

$modversion['config'][] = array(
    'name'        => 'indexscan_illegalfiles',
    'title'       => '_MI_INDEXSCAN_ILLEGALFILETYPES',
    'description' => '_MI_INDEXSCAN_ILLEGALFILETYPES_DESC',
    'formtype'    => 'textarea',
    'valuetype'   => 'text',
    'default'     => 'php|html|htm|jpg|png|gif|js|ico|txt|css|htaccess|eot|sql|swf|tpl|ttf'
);

$modversion['config'][] = array(
    'name'        => 'exep_01',
    'title'       => '_MI_INDEXSCAN_EXEP1',
    'description' => '_MI_INDEXSCAN_EXEP1_DESC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => ''
);

$modversion['config'][] = array(
    'name'        => 'exep_02',
    'title'       => '_MI_INDEXSCAN_EXEP2',
    'description' => '_MI_INDEXSCAN_EXEP2_DESC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => ''
);

$modversion['config'][] = array(
    'name'        => 'exep_03',
    'title'       => '_MI_INDEXSCAN_EXEP3',
    'description' => '_MI_INDEXSCAN_EXEP3_DESC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => ''
);

$modversion['config'][] = array(
    'name'        => 'exep_04',
    'title'       => '_MI_INDEXSCAN_EXEP4',
    'description' => '_MI_INDEXSCAN_EXEP4_DESC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => ''
);
